include/SDL3/SDL_main.h

```diff
@@ -140,20 +140,44 @@
  *  \endcode
  */
 
-#if defined(SDL_MAIN_NEEDED) || defined(SDL_MAIN_AVAILABLE)
+#if defined(SDL_MAIN_NEEDED) || defined(SDL_MAIN_AVAILABLE) || defined(SDL_MAIN_USE_CALLBACKS)
 #define main    SDL_main
 #endif
 
+#include <SDL3/SDL_events.h>
+
 #include <SDL3/SDL_begin_code.h>
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+typedef int (SDLCALL *SDL_AppInit_func)(int argc, char *argv[]);
+typedef int (SDLCALL *SDL_AppIterate_func)(void);
+typedef int (SDLCALL *SDL_AppEvent_func)(const SDL_Event *event);
+typedef void (SDLCALL *SDL_AppQuit_func)(void);
+
+#ifdef SDL_MAIN_USE_CALLBACKS
+/**
+ *  The prototypes for the application's callbacks.
+ *  These functions are to be supplied by the app instead of an
+ *  SDL_main function if they've defined SDL_MAIN_USE_CALLBACKS
+ *  before including SDL_main.h (but the SDL_main magic might
+ *  still get used below internally, if we're emulating a callback
+ *  mainloop on platforms that don't support it natively, but in that
+ *  case we'll provide SDL_main, not the app.
```

Comment:
This comment is very confusing. Is this the right place for a short description of the feature?

Reply:
I chopped out the rambling in 249152dfe372af4b294b256de8175b6502b14798. This needs more formal documentation, though, to be sure.

Reply:
Documentation is written in 4db2131589cff3c1820bfb8b1be55884c51191ae.